data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -115,13 +120,18 @@ int processSqsMessages() {
             sqsMessagesReceivedCounter.increment(sqsMessages.size());
 
             final Collection<ParsedMessage> s3MessageEventNotificationRecords = getS3MessageEventNotificationRecords(sqsMessages);
-
             // build s3ObjectReference from S3EventNotificationRecord if event name starts with ObjectCreated
             final List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntries = processS3EventNotificationRecords(s3MessageEventNotificationRecords);
 
             // delete sqs messages
             if (!deleteMessageBatchRequestEntries.isEmpty()) {
                 deleteSqsMessages(deleteMessageBatchRequestEntries);
```

Comment:
I think this needs to happen after the stopDynamic call. I am not sure what happens when a ChangeViz call is made against a deleted message

Reply:
That's fair. As long as we are handling the expected exception from SQS properly, then the order shouldn't matter